File: valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        
        stack<int> stk;
        char popped;
        int len = s.length();
        for (int i = 0; i < len; i++){
            
            char c = s[i];
            switch(c) {
                case '(': 
                    stk.push(c);
                    break;
                    
                case '{':
                    stk.push(c);
                    break;
                    
                case '[':
                    stk.push(c);
                    break;
                    
                case ')':
                    if (stk.empty()) return false;
                    popped = stk.top();
                    if (popped != '(') {
                        return false;
                    }
                    stk.pop();
                    break;
                    
                case '}':
                    if (stk.empty()) return false;
                    popped = stk.top();
                    if (popped != '{') {
                        return false;
                    }
                    stk.pop();
                    break;
                    
                case ']':
                    if (stk.empty()) return false;
                    popped = stk.top();
                    if (popped != '[') {
                        return false;
                    }
                    stk.pop();
                    break;
                    
                default : 
                    return false;
                   
           }
        }
        if (stk.empty()){
            return true;
        }
        return false;
            
        
        
        
    }
};

File: valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        #nesting of brackets is allowed 
        #use a stack
        #string will be length 1 and greater
        stack = []     
        for bracket in s:
            
            if (bracket == "{") or (bracket == "(") or (bracket == "["):
                stack.append(bracket)
            else: #bracket is either }, ), or ]

                if len(stack) == 0:
                    return False
                
                i = stack.pop()
                if i == "{":
                    if bracket is not "}":
                        return False
                elif i == "(":
                    if bracket is not ")":
                        return False
                elif i == "[":
                    if bracket is not "]":
                        return False
                #we have a matching bracket set

        if len(stack) != 0:
            return False
        else: 
            return True       
                    
                
            
            
            
            
            
            
        